openstack/compute/v2/servers/requests.go

```diff
@@ -182,6 +182,7 @@ type CreateOpts struct {
 
 // ToServerCreateMap assembles a request body based on the contents of a CreateOpts.
 func (opts CreateOpts) ToServerCreateMap() (map[string]interface{}, error) {
+	var serverCreateMapError error
```

Comment:
> I don't follow: In what way could we create a fix that would be non-unique to Gophercloud?

If the code were assign a value of "THIS-IMAGE-DOES-NOT-EXIST" to `ImageRef`, its purpose would strictly be as a dummy placeholder to fix how Gophercloud implemented bfv support.

IMO, this kind of fix is more of a meta-fix, building gophercloud's logic on top of itself, rather than being a clean abstraction to the API.

tl;dr: it feels hacky.

> Of course the most straightforward solution would be to not call servers.ToServerCreateMap and instead put the server creation logic in bfv.ToServersCreateMap (yes, resulting in code duplication...).

Indeed. I would support and be happy to implement either 1. or this option. I'm also more tha happy to leave the decision of which to you ;)